### ISSUE TITLE
Hooks feature: seems not working properly

### ISSUE THREAD

**Issue by stonebuzz** [mkdocs/mkdocs#3161]
Hello,

I have problem about the implementation of  ```on_env``` hook to add a my own ```jinja2``` filter .

There is my ```mkdocs.yml``` configuration : 

```yaml
hooks:
    - hooks.py
```

The contents of the file ```hooks.py```

```python
def say_hello(s:str):
    return "<i>Hello %s</i>" % s

def on_env(env, config, files, **kwargs):
    env.filters['say_hello'] = say_hello
    return env

```

From a ```jinja2``` template the new filter is called like this

```twig
{{nav_item.title|say_hello}}
```
After running ```mkdocs``` command (```mkdocs serve```) I get the following error

```
jinja2.exceptions.TemplateAssertionError: No filter named 'say_hello'.
```

Complete ```stacktrace```
```
INFO     -  Building documentation...
INFO     -  Cleaning site directory
INFO     -  Building en documentation
ERROR    -  Error building page 'index.md': No filter named 'say_hello'.
Traceback (most recent call last):
  File "/home/dev/.local/bin/mkdocs", line 8, in <module>
    sys.exit(cli())
  File "/usr/lib/python3/dist-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3/dist-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python3/dist-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python3/dist-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/home/dev/.local/lib/python3.10/site-packages/mkdocs/__main__.py", line 234, in serve_command
    serve.serve(dev_addr=dev_addr, livereload=livereload, watch=watch, **kwargs)
  File "/home/dev/.local/lib/python3.10/site-packages/mkdocs/commands/serve.py", line 83, in serve
    builder(config)
  File "/home/dev/.local/lib/python3.10/site-packages/mkdocs/commands/serve.py", line 76, in builder
    build(config, live_server=live_server, dirty=dirty)
  File "/home/dev/.local/lib/python3.10/site-packages/mkdocs/commands/build.py", line 332, in build
    config.plugins.run_event('post_build', config=config)
  File "/home/dev/.local/lib/python3.10/site-packages/mkdocs/plugins.py", line 522, in run_event
    result = method(**kwargs)
  File "/home/dev/.local/lib/python3.10/site-packages/mkdocs_static_i18n/plugin.py", line 589, in on_post_build
    _build_page(file.page, config, files, nav, env, dirty)
  File "/home/dev/.local/lib/python3.10/site-packages/mkdocs/commands/build.py", line 231, in _build_page
    output = template.render(context)
  File "/home/dev/.local/lib/python3.10/site-packages/jinja2/environment.py", line 1301, in render
    self.environment.handle_exception()
  File "/home/dev/.local/lib/python3.10/site-packages/jinja2/environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "/home/dev/faq/overrides/home.html", line 1, in top-level template code
    {% extends "main.html" %}
  File "/home/dev/.local/lib/python3.10/site-packages/material/main.html", line 4, in top-level template code
    {% extends "base.html" %}
  File "/home/dev/.local/lib/python3.10/site-packages/material/base.html", line 148, in top-level template code
    {% block site_nav %}
  File "/home/dev/.local/lib/python3.10/site-packages/material/base.html", line 156, in block 'site_nav'
    {% include "partials/nav.html" %}
  File "/home/dev/.local/lib/python3.10/site-packages/material/partials/nav.html", line 27, in top-level template code
    {% include "partials/nav-item.html" %}
  File "/home/dev/.local/lib/python3.10/site-packages/jinja2/environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "/home/dev/faq/overrides/partials/nav-item.html", line 32, in template
    {{nav_item.title|say_hello}}
jinja2.exceptions.TemplateAssertionError: No filter named 'say_hello'.
```

If i change ```mkdocs.yml``` like this

```yaml
hooks:
    - hook.py  # 'hook' instead of 'hooks'
```
I get the following warning

```
ERROR    -  Config value 'hooks': The path '/home/user/dev/faq//hook.py' isn't an existing file.
Aborted with 1 Configuration Errors!
```

I conclude that the original configuration file is correct and that it correctly loads the ```hooks.py``` file

The declaration of  ```on_env``` hook seems correct (IMHO)  I follow the docs from  ```Mkdocs``` and ```jinja2```
https://www.mkdocs.org/user-guide/configuration/#hooks
https://www.mkdocs.org/dev-guide/plugins/#on_env
https://jinja.palletsprojects.com/en/latest/api/#custom-filters

My ```Mkdocs``` setup information

```mkdocs``` ```1.4.2```
```mkdocs-material``` ```5.4.0```
```mkdocs-static-i18n``` ```0.53```

Best regards

**Comment by ultrabug** [comment]
Hello,

You are affected by a [mkdocs-static-i18n bug](https://github.com/ultrabug/mkdocs-static-i18n/pull/183) which is fixed but does not have a tagged release yet I'm afraid.

This has nothing to do with mkdocs.

**Comment by stonebuzz** [comment]
Hi @ultrabug 

Thank you for your quick response.

I confirm, the version ```0.53``` of ```mkdocs-static-i18n``` plugin impacts the ```hooks``` operation.

Best regards

**Comment by ultrabug** [comment]
@stonebuzz I released version 0.54 for you. You only have to update your mkdocs-static-i18n tag to 0.54 and it should work.

If not, open an issue on the plugin's project. Cheers.

**Comment by stonebuzz** [comment]
@ultrabug 

the new version works perfectly !!

Thanks again
